### DISCUSSION TITLE
Refining formatting of stimuli without using <span>

### DISCUSSION THREAD

**Discussion by CarolynBuck-Gengler** [jspsych/jsPsych#831] [Q&A] [answered]
In the workshop as well as in the RT tutorial, special formatting of the stimuli is done in the trial definition itself by surrounding the stimulus text with `<span>` (or `<div>`) and setting the style within. (e.g., 

```js
stimulus: '<div style="font-size:60px;">my stimulus text</div>' )
```

In the workshop, Becky also said one could do it in the CSS (or in a <style> element in the <head>). I did an inspect element and discovered that the id (for a html-keyboard-response) is jspsych-html-keyboard-response-stimulus. I like putting styling there so that I don't have to have it in every stimulus definition and/or create a separate property with the 'clean' version of the stimulus (as we did in the lexical decision example in the workshop), but sometimes might want to change the formatting depending on the trial. For something like the fixation cross, where the stimulus is always the same and I'm not going to analyze its data, I can always do the styling right there and have it supersede the main styling. 

But what I'm wondering is, is there a way to attach a class to each trial without having to put it in the stimulus text itself? and then style that class in the css/<style> section? Or is there another way to go about this?
(So that I could, for instance, have some be red and some be blue, without having to put a span around the stimulus text.) I would think that this would lead to the listing of trial stimuli being less susceptible to typos, as well.

(It may be that this is simple and I just haven't gotten there yet in my explorations.)

thanks!
Carolyn

**Comment by vijaymarupudi** [comment, marked as answer]
I would recommend making functions that do that for you.

In this case

```js
function largeText(input) {
  return '<div class="large-text">' + input + '</div>'
}
```

You can use it like 

```js
stimulus: largeText('my stimulus text')
```

and the styles for the `large-text` class would apply.

These can depend on stimulus parameters like

```js
stimulus: stimulusTextSize === "large" ? largeText(text) : text
```

**Reply by CarolynBuck-Gengler** [reply]
Thank you, I finally got a chance to try this out, and until Josh's enhancement below is available, this will do!
C

**Comment by jodeleeuw** [comment]
This is a design issue with the library that we've been thinking about for some time. I think we're likely to add a feature in the future that adds a custom class to the display element so that styles can be specified in a `<style>` tag or stylesheet.

The feature might look like this:

```js
var trial = {
	type: 'html-keyboard-response',
	stimulus: 'apple',
	css_classes: ['large-text']
}
```

And that would add the class `.large-text` to the root HTML element of jsPsych for that trial. Then one could specify CSS rules that applied to that class.

**Reply by CarolynBuck-Gengler** [reply]
this will be great when you get it in! thanks! C
